internal/ethapi/api.go

```diff
@@ -502,6 +502,53 @@ func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address common.Add
 	return (*hexutil.Big)(state.GetBalance(address)), state.Error()
 }
 
+// GetProof returns the Merkle-proof for a given account and optionally some storage keys.
+func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Address, storageKeys []string, blockNr rpc.BlockNumber) (map[string]interface{}, error) {
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, blockNr)
+	if state == nil || err != nil {
+		return nil, err
+	}
+
+	storageTrie := state.StorageTrie(address)
+	storageHash := types.EmptyRootHash
+	codeHash := state.GetCodeHash(address)
+	storageProof := make([]map[string]interface{}, len(storageKeys))
+
+	if storageTrie != nil {
+		storageHash = storageTrie.Hash()
+	} else {
+		codeHash = crypto.Keccak256Hash(nil)
```

Comment:
I don't follow... So if there is a `nil` storage, then you set the `codeHash` to the `nil`-hash? Why? If this is correct, and I'm not saying it isn't, then you should document why this is the case.

Reply:
I added some comments to explain it, but this is correct since `eth_getCode` for a nonexistent account would return a empty bytearray, the codehash of such a account must be `crypto.Keccak256Hash(nil)`.